build.sbt

```diff
@@ -84,17 +84,38 @@ lazy val zioConfigRefined =
     )
     .dependsOn(zioConfig % "compile->compile;test->test")
 
+lazy val runAllExamples = taskKey[Unit]("Run all main classes in examples module")
+
 lazy val examples = module("zio-config-examples", "examples")
   .settings(
     skip in publish := true,
     fork := true,
     libraryDependencies ++= Seq(
-      "eu.timepit"     %% "refined"  % refinedVersion,
-      "com.propensive" %% "magnolia" % magnoliaVersion
-    )
+      "eu.timepit"            %% "refined"    % refinedVersion,
+      "com.propensive"        %% "magnolia"   % magnoliaVersion,
+      "com.github.pureconfig" %% "pureconfig" % "0.12.3"
+    ),
+    runAllExamples :=
+      Def
+        .taskDyn({
+          val c    = (discoveredMainClasses in Compile).value
+          val runs = (runMain in Compile)
+
+          val x = c.map(cc => {
+            Def.task {
+              runs.toTask(s" ${cc}").value
+            }
+          })
+
+          Def.sequential(x)
+        })
+        .value
```

Comment:
All of the examples has assertions to provide the users a view of the return types + validating the examples itself.  For contributors, `runAllExamples` will run all assertions in the examples quickly in parallel